tests/glsl-intrinsic/subpass-input/subpass-input-hlsl.slang

```diff
@@ -0,0 +1,32 @@
+//TEST:SIMPLE(filecheck=CHECK_GLSL): -target glsl -stage fragment -entry main -allow-glsl
+//TEST:SIMPLE(filecheck=CHECK_SPV): -target spirv -emit-spirv-directly -stage fragment -entry main -allow-glsl
+//TEST:SIMPLE(filecheck=CHECK_HLSL): -target hlsl -stage fragment -entry main -allow-glsl
+
+// CHECK_SPV-DAG: OpEntryPoint
+// CHECK_SPV-DAG: InputAttachmentIndex 3
+// CHECK_SPV-DAG: InputAttachmentIndex
+
+// CHECK_GLSL-DAG: void main()
+// CHECK_GLSL-DAG: input_attachment_index = 3
+// CHECK_GLSL-DAG: input_attachment_index
+
+// CHECK_HLSL-DAG: main()
+// CHECK_HLSL-DAG: [vk::input_attachment_index(3)]
+// CHECK_HLSL-DAG: vk::input_attachment_index
+[[vk::input_attachment_index(3)]] SubpassInput subpassHLSL1;
+SubpassInput subpassHLSL2;
+
+RWTexture2D<float4> t;
+
+layout (location = 0) out vec4 outColor;
+
+void main() {
+// CHECK_SPV-COUNT-2: OpImageRead
+// CHECK_GLSL-COUNT-2: subpassLoad
+// CHECK_HLSL-COUNT-2: SubpassLoad
+    outColor = vec4(true
+        && subpassHLSL1.SubpassLoad() == vec4(1)
```

Comment:
please stop using && in all of your test code, because I'm afraid that the short-circuit could bypass something, though I know that the short-circuit is not a default behavior. But in case some day to enable it by default, and people forgot about changing the test.

You can use "&" instead.

Reply:
Booleans are 32 bit in GLSL/HLSL/Metal/Other.
GLSL says the result of a boolean expression is 0 or 1, but I don't think this is ensured for other targets.
`0b10` is true, `0b01` is true, `0b10 & 0b01` is false, and `0b10 && 0b01` is true.

I am confused on how short-circuit would break the code, if the code returns false (short-circuits), then the test will fail as it should?

Reply:
The whole point is that I want to remind you that if you want each statement to be executed,
&& is not gonna ensure that.

I just want to remind you to be careful with "&&" in your test code.

If you worry about boolean representation issue, you can also use if..else or even !!(expression).